File: minSpantree.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;


class Vertex {

    public int value;
    public Vertex parent;

    public Vertex(int value) {
        this.value = value;
        this.parent = this;
    }

    public Vertex(int value, Vertex parent) {
        this.value = value;
        this.parent = parent;
    }

    public Vertex getRoot() {
        return this.parent == this ? this : this.parent.getRoot();
    }

}
class Edge implements Comparable<Edge>{

    public Vertex u, v;
    public int weight;

    Edge(Vertex u, Vertex v, int weight)
    {
        this.u = u;
        this.v = v;
        this.weight = weight;
    }

    public int compareTo(Edge e) {
        return this.weight - e.weight;
    }

}

class Main {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        for(int test = scanner.nextInt(); test > 0; test--) {

            int numVertices = scanner.nextInt();
            int numEdges = scanner.nextInt();

            Vertex[] vertices = new Vertex[numVertices];
            ArrayList<Edge> edges = new ArrayList<Edge>();

            for(int i = 0; i < numEdges; ++i) {

                int uValue = scanner.nextInt() - 1;
                int vValue = scanner.nextInt() - 1;

                if (vertices[uValue] == null) {
                    vertices[uValue] = new Vertex(uValue);
                }

                if (vertices[vValue] == null) {
                    vertices[vValue] = new Vertex(vValue);
                }

                edges.add( new Edge( vertices[uValue], vertices[vValue], scanner.nextInt() ) );

            }

            Collections.sort(edges);

            int totalWeight = 0;

            for (Edge edge : edges) {
                if (edge.u.getRoot() != edge.v.getRoot()) {
                    edge.v.getRoot().parent = edge.u.getRoot();
                    totalWeight += edge.weight;
                }
            }

            System.out.println(totalWeight);

        //
        }
    }
}

File: minSpantree.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var stdIn = "",
    inputData = [];

process.stdin.on('data', function (data) {
    stdIn += data;
});

process.stdin.on('end', function () {
    inputData = stdIn.split(/[\s\r\n]+/g);
    minSpantree();
});

function nextInt() {
    return parseInt(inputData.shift());
}

function Vertex(value) {
    return {
        "value": value,
        "parent": null,
        "getRoot": function () {
            return this.parent == null ? this : this.parent.getRoot()
        }
    }
}

function minSpantree() {

    for (var test = nextInt(); test > 0; test--) {

        var numVertices = nextInt();
        var numEdges = nextInt();

        var vertices = [];
        for (var i = 0; i < numVertices; i++) {
            vertices[i] = new Vertex(i);
        }

        var edges = [];

        for (var i = 0; i < numEdges; i++) {

            var uValue = nextInt() - 1;
            var vValue = nextInt() - 1;

            edges.push({
                "u": vertices[uValue],
                "v": vertices[vValue],
                "weight": nextInt()
            });

        }

        edges.sort(function (a, b) {
            return a.weight - b.weight;
        })

        var totalWeight = 0;

        edges.forEach(function (edge) {
            if (edge.u.getRoot() != edge.v.getRoot()) {
                edge.v.getRoot().parent = edge.u.getRoot();
                totalWeight += edge.weight;
            }
        });

        console.log(totalWeight);

    }

}